src/App.js

```diff
@@ -24,62 +24,87 @@ import getToken from './lib/tokens';
 
 function App() {
   const [listId, setListId] = useState(null);
-  const [userToken, setUserToken] = useState('');
 
-  function saveToken(e) {
-    e.preventDefault();
+  function createList() {
     const token = getToken();
-    localStorage.setItem('token', token);
-    setUserToken(token);
+
+    return db
+      .collection('lists')
+      .add({ token: token })
+      .then((results) => {
+        const { id } = results;
+        if (id === undefined) throw new Error('Failed to create list.');
+        return id;
+      })
+      .then((newListId) => {
+        // list successfully created in db, so update listId state and save token to localStorage
+        setListId(newListId);
```

Comment:
For my own understanding, I assume you are de-structuring the list's id out of the results from Firestore, then using that as newListId and setting that as listId? A few comments here might have been helpful for me personally!

Reply:
Good suggestion! I've made a note to add some in.

Reply:
I agree with Connie as far as adding some comments. I'm also thinking along the lines of what if this was a work situation. Where maybe someone would have to come behind you blindly without knowing the AC or the codebase already. They maybe a little confused here. However, I do really think that the if statement is pretty slick because of the "throw". I never really see it that often, and I feel like you probably eliminated several lines of code just by using that. Really good job here.